includes/class-wc-tracker.php

```diff
@@ -445,12 +446,23 @@ private static function get_orders() {
 			$orders_count = count( $orders );
 		}
 
-		$order_data['first']            = gmdate( 'Y-m-d H:i:s', $first );
-		$order_data['last']             = gmdate( 'Y-m-d H:i:s', $last );
-		$order_data['processing_first'] = gmdate( 'Y-m-d H:i:s', $processing_first );
-		$order_data['processing_last']  = gmdate( 'Y-m-d H:i:s', $processing_last );
+		if ( $first !== $first_time ) {
+			$order_data['first'] = gmdate( 'Y-m-d H:i:s', $first );
+		}
+
+		if ( $processing_first !== $first_time ) {
+			$order_data['processing_first'] = gmdate( 'Y-m-d H:i:s', $processing_first );
+		}
+
+		if ( $last ) {
+			$order_data['last'] = gmdate( 'Y-m-d H:i:s', $last );
+		}
+
+		if ( $processing_last ) {
+			$order_data['processing_last']  = gmdate( 'Y-m-d H:i:s', $processing_last );
+		}
 
-		return $order_data;
+		return array_map( 'strval', $order_data );
```

Comment:
Based on [this WP core ticket](https://core.trac.wordpress.org/ticket/42918), it might be better to use `(string)` type casting for type conversion.

Reply:
Good point, I haven't noticed that. Either way it's fine for me, it's a small difference.